File: Src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
# import ros 
import rclpy 
from rclpy.node import Node ## import  node class
from example_interfaces.msg import String # dependencies

class RobotNewsStationNode(Node):

    def __init__(self):
        super().__init__("robot_news_station")

        self.publisher_ = self.create_publisher(String,"robot_news", 10) # create publisher with string data type and with a topic namde in this node , here 10 is a que size as like data buffer. We got publisher_ object 
        self.timer_= self.create_timer(0.5, self.publish_news)
        self.get_logger().info("Robot news Station Has been started")

    def publish_news(self):
        msg = String()
        msg.data = "hello"
        self.publisher_.publish(msg)
        
def main(args=None):
    rclpy.init(args=args) ## initialize ros to create ros communiation 
    node = RobotNewsStationNode()
    rclpy.spin(node) # spin allow to node
    rclpy.shutdown() # ros communication shutdown

if __name__== "__main__":
    main()

File: src/my_cp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewsStationNode: public rclcpp::Node
{
public:
    RobotNewsStationNode(): Node("robot_news_station"), robot_name_("R2D2")
    {
        publisher_=this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500),
                                         std::bind(&RobotNewsStationNode::publishNews, this)); // std::bind use for callback 
       RCLCPP_INFO(this->get_logger(), "robot news station node start");

    } 

private:
void publishNews()
{
  auto msg = example_interfaces::msg::String();
  msg.data = std::string("Hi, this is ") + robot_name_ + std::string(" from the Robot News Station");
  publisher_->publish(msg);

}

std::string robot_name_;
rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_; // declare publisher before creating with sharedptr
rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
